external/model-preparation-algorithm/mpa_tasks/utils/data_utils.py

```diff
@@ -44,34 +44,48 @@ def is_training_subset(subset):
         return subset.name in ["TRAINING", "VALIDATION"]
 
     def load_image_from_cache(filename: str, to_float32=False):
-        with open(filename, "rb") as f:
+        try:
+            f = open(filename, "rb")
             fcntl.flock(f, fcntl.LOCK_SH)
-            try:
-                cached_img = cv2.imread(filename)
-                if to_float32:
-                    cached_img = cached_img.astype(np.float32)
-                return cached_img
-            except Exception as e:
-                logger.warning(f"Skip loading cached {filename} \nError msg: {e}")
-            finally:
+            cached_img = np.asarray(bytearray(f.read()))
+            fcntl.flock(f, fcntl.LOCK_UN)
+            f.close()
```

Comment:
I think it's better to separate fcntl.flock and file open part as each `try except`.